presto-bytecode/src/main/java/com/facebook/presto/bytecode/ClassDefinition.java

```diff
@@ -266,6 +270,13 @@ public MethodDefinition declareMethod(
             ParameterizedType returnType,
             Iterable<Parameter> parameters)
     {
+        int parameterCount = Iterables.size(parameters);
+        checkArgument(
+                parameterCount > MAX_ALLOWED_VARARG_ARGUMENTS,
+                "Number of arguments (%d) for vararg functions exceeds limit (%d)",
```

Comment:
Code generators uses declareMethod() to create vararg method runtime. (i.e. ArrayConsturctor.java). But yep, I should move this into constructor of MethodDefinition